### PR TITLE
feature/connect api

### PR DESCRIPTION
**Pull request by sazanik**

### CODE REVIEW

File: src/store/users/actions.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  email: string;
+  password: string;
+}
+
+const sigInRequest = async ({
+  auth,
+  email,
+  password,
+}: Props): Promise<void> => {
+  await signInWithEmailAndPassword(auth, email, password);
```
Comment (ArturRusak) [inline src/store/users/actions.ts:19]: bad name

File: src/components/DialogBox/DialogBox.tsx [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+export const DialogBox = ({
+  isOpenDialog,
+  setIsOpenDialog,
+  handleClickDelete,
+  entity,
+}: Props): JSX.Element => {
+  const { t } = useTranslation();
+
+  const dialogsTranslation = {
```
Comment (ArturRusak) [inline src/components/DialogBox/DialogBox.tsx:27]: pass it as props

File: src/components/Table/Table.tsx [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+      pageSize={13}
+      components={{
+        Toolbar: () =>
+          TableToolbar({
+            entity,
+            setEdit,
+            setIsOpenModal,
+          }),
+      }}
+      rowsPerPageOptions={[13]}
```
Comment (ArturRusak) [inline src/components/Table/Table.tsx:38]: move to named const

File: src/context/AuthContextProvider.tsx [hunk 4]
```diff
@@ -10,15 +12,16 @@ interface Props {
...
   const signIn = (newToken: string, cb: () => void): void => {
     setToken(newToken);
     cb();
   };
 
-  const logOut = (cb: () => void): void => {
+  const logOut = (): void => {
     setToken(null);
-    cb();
+    auth.signOut().catch((e) => console.error(e));
```
Comment (ArturRusak) [inline src/context/AuthContextProvider.tsx:24]: should be passed to store

File: src/screens/BookCard/ScreensBookCard.tsx [hunk 5]
```diff
@@ -22,6 +22,12 @@ export const ScreensBookCard = (): JSX.Element => {
     booksSelectors.selectById(store.getState(), id)
   );
 
+  const getAuthorName = (): string | undefined => {
+    if (!book) return;
```
Comment (ArturRusak) [inline src/screens/BookCard/ScreensBookCard.tsx:26]: Don't write in one lien

File: src/types/enums.ts [hunk 6]
```diff
@@ -5,6 +5,8 @@ export enum Actions {
 }
 
 export enum Fields {
+  FirstName = 'firstName',
```
Comment (ArturRusak) [inline src/types/enums.ts:8]: use capital letters with Enums

**Review by ArturRusak — CHANGES_REQUESTED** [review]

File: src/store/users/actions.ts (hunk 1 above)
Comment (sazanik) [reply src/store/users/actions.ts:19]: it is method firebase API

File: src/components/Table/Table.tsx (hunk 3 above)
Comment (sazanik) [reply src/components/Table/Table.tsx:37]: fixed

File: src/components/DialogBox/DialogBox.tsx (hunk 2 above)
Comment (sazanik) [reply src/components/DialogBox/DialogBox.tsx:27]: fixed

File: src/screens/BookCard/ScreensBookCard.tsx (hunk 5 above)
Comment (sazanik) [reply src/screens/BookCard/ScreensBookCard.tsx:26]: fixed

File: src/types/enums.ts (hunk 6 above)
Comment (sazanik) [reply src/types/enums.ts:8]: fixed

File: src/context/AuthContextProvider.tsx (hunk 4 above)
Comment (sazanik) [reply src/context/AuthContextProvider.tsx:24]: fixed

File: src/store/app/appSlice.ts [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+};
+
+export const appSlice = createSlice({
+  name: 'app',
+  initialState,
+  reducers: {
+    setLoading: (state) => {
+      state.loading = true;
+    },
+    stopLoading: (state) => {
```
Comment (ArturRusak) [inline src/store/app/appSlice.ts:48]: why do we need this state? 
also, why do you implement business logic here?

File: src/store/authors/actions.ts [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+  return (await getDoc(docRef)).data() as AuthorProps;
+};
+
+const removeDoc = async (id: string): Promise<string | undefined> => {
+  const docRef = doc(db, 'authors', id);
+  await deleteDoc(docRef);
+  return docRef.id;
+};
+
+const readDocs = async (): Promise<AuthorProps[]> => {
```
Comment (ArturRusak) [inline src/store/authors/actions.ts:38]: Write it together with createAsyncThunk or create fabric function.

File: src/store/authors/authorsSlice.ts [hunk 9]
```diff
@@ -1,30 +1,48 @@
...
+  reducers: {},
+  extraReducers: (builder) => {
+    builder
+      .addCase(createAuthor.fulfilled, (state, action) => {
+        const { payload: author } = action;
+        authorsAdapter.addOne(state, <AuthorProps>author);
+      })
+      .addCase(updateAuthor.fulfilled, (state, action) => {
+        const { payload: author } = action;
+        if (author) {
```
Comment (ArturRusak) [inline src/store/authors/authorsSlice.ts:28]: Why do you check here author ?

File: src/store/authors/authorsSlice.ts [hunk 10]
```diff
@@ -1,30 +1,48 @@
...
+            id: author.id,
+            changes: {
+              ...author,
+            },
+          });
+        }
+      })
+      .addCase(removeAuthor.fulfilled, (state, action) => {
+        const { payload: id } = action;
+        if (id) {
```
Comment (ArturRusak) [inline src/store/authors/authorsSlice.ts:39]: the same

**Review by ArturRusak — CHANGES_REQUESTED** [review]

**Pull request merged**